### PR TITLE
airport challenge

### PR DESCRIPTION
**Pull request by hawksdoves**

### CODE REVIEW

**Comment by mechacoach** [comment]
You had your airport_challenge reviewed by **Matthew Bridges**

### The good points are:

* Overview of project in the README
* `context` or `describe` blocks are used (where appropriate) to group tests
* `before` blocks are used where appropriate
* There are a minimal number of untested paths in the code
* Unit tests all have single expectations
* The code follows Ruby naming conventions, Readability is supported by consistent naming
* Guard clauses are used where appropriate, Booleans are returned implicitly where appropriate, removeallcommented-outcode: There is no commented-out code in the submission
* Code avoids unnecessary public methods, Code avoids public access to internal objects (e.g. planes array)
* Classes have single responsibilities, Each methods has a single responsibility
* Code avoids strings where it should use symbols
* Command/Query naming is not an issue in the code
* The solution does not have any redundant code

### You should consider the following improvements:

* Installation Instructions should be in the README
* Use named subject with `described_class`
* Avoid Vacuous tests
* Handle randomness in tests
* Get rid of redundant `respond_to` expectations
* Remove all commented-out code
* Avoid magic numbers (e.g. on capacity)
* Improper use of attr_*

### Additional comments:
ObjectSpace wizardry, cool code., avoidmagicnumberse.g.oncapacity: Code avoids magic numbers

see https://github.com/makersacademy/airport_challenge/blob/master/docs/review.md for more details

File: docs/airport.rb [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+	def initialize(capacity=DEFAULT_CAPACITY)
+		@capacity = capacity
+		@storage = []
+		@weather = Weather.new
+	end
+
+	def in_airport?(plane)
+		storage.include?(plane)
+	end
```
Comment (danldb) [inline docs/airport.rb]: ARGH my eyes! please fix your indentation. ;-)

File: docs/airport.rb [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+			plane.landed
+		else
+			announcement
+		end
+	end
+
+	def releases_plane(plane)
+		if airport_open?	
+			fail 'This plane is already in the air, so it cannot take off.' if plane.plane_status		
+			fail 'This plane is not at this airport, so it cannot take off from here.' if in_airport?(plane) == false
```
Comment (danldb) [inline docs/airport.rb]: prefer:

`fail 'This plane is not at this airport, so it cannot take off from here.' if in_airport?(plane)`

(remembering that in_airport? returns a boolean already)

File: docs/airport.rb [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+			storage.delete(plane)
+			plane.flying
+		else
+			announcement
+		end
+	end
+
+	def plane_location(plane)
+		plane.plane_status ? 'This plane is flying.' : find_plane(plane)
```
Comment (danldb) [inline docs/airport.rb]: Looks like feature envy - everything about this method regards the plane, so the responsibility is probably on the plane, not the airport.

File: docs/airport.rb [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+		found =	airports.select {|airport| airport.in_airport?(plane)}
+		found[0]
+	end
+
+	def full?
+		storage.length>=capacity 
+	end
+
+	def airport_open?
+	 	weather.random_weather == "not stormy"
```
Comment (danldb) [inline docs/airport.rb]: prefer symbols to strings

File: docs/plane.rb [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+	def take_off(from)
+		from.releases_plane(self)
+	end
+
+	def land(at)
+		at.accepts_plane(self)
+	end
+	
+	def plane_status
```
Comment (danldb) [inline docs/plane.rb]: consider this method name. plane_status returns true or false. If I asked a plane it's status, would I expect it to say `true`

**Comment by danldb** [comment]
Not bad. Broken YAGNI with the plane location thing. Slightly over-complicated, and more elements than were needed to solve the problem.

**Comment by danldb** [comment]
also indentation is horrible. I would recommend you change the settings in your text editor - use spaces instead of tabs, 2 spaces per indent. Really important that you focus on this and get it good, bad indentation will prevent you from getting job interviews!

**Pull request closed without merging**